### PR TITLE
Sync Experiments in Explorer and Viewer

### LINKED ISSUE #140: Better handle multiple open traces
It is currently possible to open multiple traces separately, but the trace explorer is not in sync. with the trace container in focus.

To Reproduce:

- Use "Open Trace(s)" menu to open trace1 -> trace1 is opened in trace container
- Open Trace Explorer
- Add a view -> view added and populated
- Go back to File Explorer and use "Open Trace(s)" menu to open trace2 -> trace is opened in trace container
- Select Trace Explorer
- Select trace container for trace2
- Open view -> view is added to trace container of **trace 1** and not for **trace 2**

Looking closely at the Trace Explorer, then the trace 1 is selected and the list of available analysis of trace 1 shown.

When selecting trace container for trace 2 (set focus) doesn't update the selection in the Trace Explorer to trace 2 as well as the available analysis are not updated. 

The trace container in focus and the content of the Trace Explorer need to be in sync.

### PR DESCRIPTION
**Pull request by muddana-satish**
fixes #140

Signed-off-by: muddana-satish <satish.muddana@ericsson.com>

### CODE REVIEW

**Comment by muddana-satish** [comment]
Here are the scenarios I have tested:

Viewer -> Explorer:

Kapture 2020-11-05 at 13 17 08

Explorer -> Viewer:

Kapture 2020-11-05 at 13 22 03


Refresh:

It always selects the first experiment in the explorer. But will retain the correct experiment after everything is loaded. Not a good UX. Any thoughts on this?

Kapture 2020-11-05 at 13 27 09

SplitWindow:

Only one experiment is selected in the explorer though there can be multiple tabs that can be visible because of split window. By default we always select the last window's active experiment.

Kapture 2020-11-05 at 13 32 48

Experiment Added:
Kapture 2020-11-05 at 13 46 37

Experiment Closed:
Kapture 2020-11-05 at 13 49 21


Please let me know if there are any other scenarios that I missed.

**Comment by muddana-satish** [comment]
Note: I have not covered any of the issues mentioned in #149. I will create a new PR for that.

File: viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { EditorManager, EditorOpenerOptions } from '@theia/editor/lib/browser';
 import URI from '@theia/core/lib/common/uri';
 import { Experiment } from 'tsp-typescript-client/lib/models/experiment';
 import { ExperimentManager } from '../../common/experiment-manager';
+import { TraceViewerWidget } from '../trace-viewer/trace-viewer';
```
Comment (tahini) [inline viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx:15]: Just highlighting this is adding a circular dependency between trace-viewer and trace-explorer-widget. It is definitely not a good practice in general, even though it works here.

With the split in packages and vscode extension, this is the kind of interaction that may well be re-architectured (how signals are sent between components, etc), so you don't have to try to fix it just yet. But could you add a FIXME above this line, just so we can come back to this eventually?

**Review by tahini — APPROVED** [review]
Works nicely, thanks!

**Comment by mirsky-work** [comment]
Works great, thanks!

Can we have a visual indication which tab is active in a split window?
active_tab

File: viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx (hunk 1 above)
Comment (muddana-satish) [reply viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx:15]: Thank you for sharing this.

**Comment by muddana-satish** [comment]
> Can we have a visual indication which tab is active in a split window?

The focus on tabs in other theia-apps is working as expected (highlights the active tab name). But for some reason it isn't working in trace-extension. I would suggest creating a new issue for that (since it was never a working feature in trace-extension). 

Created #182. Thanks @mirsky-work for verifying and pointing out this issue.

File: viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx (hunk 1 above)
Comment (muddana-satish) [reply viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx:16]: Added a `FIXME` comment. Thanks.

**Review by bhufmann — APPROVED** [review]
Nice improvement! Thanks for the contribution.

**Pull request merged**